webhook/main.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	cloudbuild "cloud.google.com/go/cloudbuild/apiv1/v2"
+	"cloud.google.com/go/cloudbuild/apiv1/v2/cloudbuildpb"
+	kms "cloud.google.com/go/kms/apiv1"
+	"github.com/abcxyz/pkg/githubauth"
+	"github.com/google/go-github/v69/github"
+	"github.com/sethvargo/go-gcpkms/pkg/gcpkms"
+	"golang.org/x/oauth2"
+)
+
+// gcloud run deploy webhook-go --region=us-west1 --source . --update-secrets=/etc/secrets/webhook/key=${KEY_NAME}:latest --allow-unauthenticated --set-env-vars=APP_ID=${APP_ID},TRIGGER_ID=${TRIGGER_ID},PROJECT_ID=${PROJECT_ID},KEY_ID=${KEY_ID}
+
+func main() {
```

Comment:
This is a cloud run service which is long lived. To ensure proper shutdown you should use the [realMain pattern](https://github.com/abcxyz/readability/wiki/Go-Best-Practices#the-realmain-pattern) which ensures the correct signals are captured and everything gets released cleanly.